### PR TITLE
Add a creatorId to molecule and calculation sagas

### PR DESCRIPTION
**Pull request by bnmajor**
Pass in a creatorId parameter for fetchMoleculesFromGirder and
fetchCalculationsFromGirder to allow filtering calculations and molecules by user
id.

Signed-off-by: Brianna Major <brianna.major@kitware.com>

### CODE REVIEW

File: packages/sagas/src/index.js [hunk 1]
```diff
@@ -36,11 +36,11 @@ export function setPaginationDefaults(options)
   }
 }
 
-export function fetchMoleculesFromGirder(options={}) {
+export function fetchMoleculesFromGirder(options={}, creatorId) {
   // Let's modify a clone of the options instead of the original options
   const optionsClone = { ...options }
   setPaginationDefaults(optionsClone)
-  const params = { params: optionsClone }
+  const params = { params: {...optionsClone, creatorId} }
```
Comment (cjh1) [inline packages/sagas/src/index.js]: I think we only want to include the `creatorId` if its not empty. I am seeing the following error when I try to access `Molecules` ( not just my molecules ).

```
{"message": "InvalidId: InvalidId(\"'' is not a valid ObjectId, it must be a 12-byte input or a 24-character hex string\",)", "trace": ["<FrameSummary file /girder/girder/api/rest.py, line 630 in endpointDecorator>", "<FrameSummary file /girder/girder/api/rest.py, line 1191 in GET>", "<FrameSummary file /girder/girder/api/rest.py, line 947 in handleRoute>", "<FrameSummary file /mongochemserver/girder/molecules/molecules/molecule.py, line 84 in find>", "<FrameSummary file /mongochemserver/girder/molecules/molecules/models/molecule.py, line 62 in findmol>", "<FrameSummary file /usr/local/lib/python3.5/dist-packages/bson/objectid.py, line 110 in __init__>", "<FrameSummary file /usr/local/lib/python3.5/dist-packages/bson/objectid.py, line 213 in __validate>", "<FrameSummary file /usr/local/lib/python3.5/dist-packages/bson/objectid.py, line 40 in _raise_invalid_id>"], "type": "internal", "uid": "bf2ac37f-754b-4ae8-b4da-54fceb057a60"}
```

basically its try to convert '' to a mongo id.

Comment (cjh1) [reply packages/sagas/src/index.js]: We would need the same check for calculations as well ...

Comment (psavery) [reply packages/sagas/src/index.js]: By the way, I think `creatorId` is always added now on molecule and calculation creation, but you might have some that were created before that was changed...

Comment (cjh1) [reply packages/sagas/src/index.js]: Sure, the problem here, is querying molecules using `creatorId` when its '', which is not a valid `ObjectId`.

Comment (bnmajor) [reply packages/sagas/src/index.js]: Fixed to only include `creatorId` in `params` if it is not empty.

File: packages/sagas/src/index.js [hunk 2]
```diff
@@ -58,10 +63,11 @@ export function fetchMoleculeByIdFromGirder(id) {
 // Molecules
 
 export function* fetchMolecules(action) {
-  const options = action.payload
+  console.log(action);
```
Comment (cjh1) [inline packages/sagas/src/index.js]: Left over debug

File: packages/sagas/src/index.js [hunk 3]
```diff
@@ -36,11 +36,16 @@ export function setPaginationDefaults(options)
...
 
-export function fetchMoleculesFromGirder(options={}) {
+export function fetchMoleculesFromGirder(options={}, creatorId) {
   // Let's modify a clone of the options instead of the original options
   const optionsClone = { ...options }
   setPaginationDefaults(optionsClone)
-  const params = { params: optionsClone }
+
+  var params = { params: optionsClone }
+  if (creatorId){
```
Comment (cjh1) [inline packages/sagas/src/index.js]: Can we use `!isNil(...)` from lodash here.

File: packages/sagas/src/calculations.js [hunk 4]
```diff
@@ -12,11 +12,16 @@ import girderClient from '@openchemistry/girder-client';
...
 
-function fetchCalculations(options={}) {
+function fetchCalculations(options={}, creatorId) {
   // Let's modify a clone of the options instead of the original options
   const optionsClone = { ...options };
   setPaginationDefaults(optionsClone);
-  const params = { params: optionsClone };
+
+  var params = { params: optionsClone };
+  if (creatorId) {
```
Comment (cjh1) [inline packages/sagas/src/calculations.js]: Can we use !isNil(...) from lodash here.

Comment (bnmajor) [reply packages/sagas/src/calculations.js]: Fixed.

File: packages/sagas/src/index.js (hunk 3 above)
Comment (bnmajor) [reply packages/sagas/src/index.js]: Fixed.

File: packages/sagas/src/index.js (hunk 2 above)
Comment (bnmajor) [reply packages/sagas/src/index.js]: Removed.

**Pull request merged**